Updated RELEASES.md for 1.35.0 (#60546)
rust-lang/rust: RELEASES.md

- [Expanded `Debug` output (`{:#?}`) for structs now has a trailing comma on the
  last field.][59076]
- [`char::{ToLowercase, ToUppercase}` now
  implement `ExactSizeIterator`.][58778]
- [All `NonZero` numeric types now implement `FromStr`.][58717]
- [Removed the `Read` trait bounds
  on the `BufReader::{get_ref, get_mut, into_inner}` methods.][58423]
- [You can now call the `dbg!` macro without any parameters to print the file
  and line where it is called.][57847]
- [In place ASCII case conversions are now up to 4× faster.][59283]
  e.g. `str::make_ascii_lowercase`
- [`hash_map::{OccupiedEntry, VacantEntry}` now implement `Sync`
  and `Send`.][58369]

Stabilized APIs
---------------
- [`f32::copysign`]
- [`f64::copysign`]
- [`RefCell::replace_with`]
- [`RefCell::map_split`]
- [`ptr::hash`]
- [`Range::contains`]
- [`RangeFrom::contains`]
- [`RangeTo::contains`]
- [`RangeInclusive::contains`]
- [`RangeToInclusive::contains`]
- [`Option::copied`]

Cargo
-----
- [You can now set `cargo:rustc-cdylib-link-arg` at build time to pass custom
  linker arguments when building a `cdylib`.][cargo/6298] Its usage is highly
  platform specific.
  
Misc
----

> - [The 1.35.0 rust toolchain is now available natively for musl based distros.][58575]

Reviewer: ```suggestion
- [The Rust toolchain is now available natively for musl based distros.][58575]
```

Version 1.35.0 (2019-05-23)
==========================

Language
--------

> - [`FnOnce`, `FnMut`, and the `Fn` traits are now implemented for `Box<FnOnce>`,

Reviewer 2: I'm debating whether we should expand this bullet with "why should you care" information... On the one hand, it is probably the most important change this release. On the other, the bullet has all information that is sufficient from a "this is what happened" POV.

Author: This is something that could be expanded on in the blog post.

  `Box<FnMut>`, and `Box<Fn>` respectively.][59500]

> - [You can now coerce closures into unsafe functions.][59580] e.g.

Reviewer 2: ```suggestion
- [You can now coerce closures into unsafe function pointers.][59580] e.g.
```

  ```rust

>   unsafe fn call_unsafe(func: unsafe fn() -> ()) -> () {

Reviewer 3: ```suggestion
  unsafe fn call_unsafe(func: unsafe fn()) {
```

      func()
  }

  pub fn main() {
      unsafe { call_unsafe(|| {}); }
  }
  ```

Compiler
--------
- [Added the `armv6-unknown-freebsd-gnueabihf` and
  `armv7-unknown-freebsd-gnueabihf` targets.][58080]
- [Added the `wasm32-unknown-wasi` target.][59464]

Libraries
---------

> - [`Thread` will now show its id in `Debug` output.][59460]

Reviewer 4: ```suggestion
- [`Thread` will now show its ID in `Debug` output.][59460]
```

- [`StdinLock`, `StdoutLock`, and `StderrLock` now implement `AsRawFd`.][59512]
- [`alloc::System` now implements `Default`.][59451]
- [Expanded `Debug` output (`{:#?}`) for structs now has a trailing comma on the
  last field.][59076]

> - [`char::{ToLowercase, ToUppercaseImplement}` now

Reviewer: ```suggestion
- [`char::{ToLowercase, ToUppercase}` now
```
